CRM/Report/Form.php

```diff
@@ -4779,6 +4779,16 @@ protected function setOutputMode() {
       FALSE,
       CRM_Utils_Array::value('task', $this->_params)
     ));
+    if ($this->_outputMode !== NULL) {
+      // We didn't get it using output, so we now try with task.
+      $this->_outputMode = str_replace('report_instance.', '', CRM_Utils_Request::retrieve(
+          'task',
+          'String',
+          CRM_Core_DAO::$_nullObject,
+          FALSE,
+          CRM_Utils_Array::value('task', $this->_params)
+        ));
+    }
```

Comment:
Thanks for the feedback.  I am tied up until after Christmas, so any changes will have to wait till the New Year, I'm afraid.  I daren't make a quick change that does not work properly.